### PR TITLE
docs: add a note about infinite loops

### PR DESCRIPTION
**Pull request by tmatsuo**

### CODE REVIEW

File: AUTHORING_GUIDE.md [hunk 1]
```diff
@@ -347,8 +348,43 @@ is expected. Strive to verify the content of the output rather than the syntax.
...
 
+We recommend the timeout parameter to be around the number that gives
+you more than 90% success rate. Don't put too long a timeout.
+
+Now this test is inevitably flaky, so consider marking the test as
+`flaky` as follows:
+
+```python
+
+@pytest.mark.flaky(max_runs=3, min_passes=1)
```
Comment (kurtisvg) [inline AUTHORING_GUIDE.md:378]: Wasn't there an issue with pytest and flaky? Do we want to recommend backoff instead?

Comment (tmatsuo) [reply AUTHORING_GUIDE.md:378]: What was the issue? I heard this several times, but I'm not sure what the issue was.

Comment (tmatsuo) [reply AUTHORING_GUIDE.md:378]: `flaky` gives you a test report, something like:

```
===Flaky Test Report===

test_counter failed (9 runs remaining out of 10).
        <class 'Exception'>
        not yet
        [<TracebackEntry /usr/local/google/home/tmatsuo/work/tmp/test.py:12>]
test_counter failed (8 runs remaining out of 10).
        <class 'Exception'>
        not yet
        [<TracebackEntry /usr/local/google/home/tmatsuo/work/tmp/test.py:12>]
test_counter passed 1 out of the required 1 times. Success!

===End Flaky Test Report===

```

I think it's useful when we want to scan these logs.

Comment (kurtisvg) [reply AUTHORING_GUIDE.md:378]: I thought you had mentioned it, but if it's working well ignore me :)

**Pull request merged**